shared/src/main/scala/tastyquery/ast/Symbols.scala

```diff
@@ -243,18 +243,128 @@ object Symbols {
           case _ => None
         mySignature = sig
         sig
-  }
 
-  object RegularSymbol:
-    private[tastyquery] def create(name: Name, owner: Symbol): RegularSymbol =
-      RegularSymbol(name, owner)
+    /** If this symbol has a `MethodicType`, returns a `SignedName`, otherwise a `Name`. */
+    final def signedName(using Context): Name =
+      signature.fold(name) { sig =>
+        val name = this.name.asSimpleName
+        val targetName = name // TODO We may have to take `@targetName` into account here, one day
+        SignedName(name, sig, targetName)
+      }
+
+    final def paramSymss(using Context): List[Either[List[TermSymbol], List[LocalTypeParamSymbol]]] =
+      tree match
+        case Some(ddef: DefDef) =>
+          ddef.paramLists.map {
+            case Left(params)   => Left(params.map(_.symbol))
+            case Right(tparams) => Right(tparams.map(_.symbol.asInstanceOf[LocalTypeParamSymbol]))
+          }
+        // TODO: java and scala 2 methods do not have trees
+        case _ => Nil
+  end TermSymbol
+
+  object TermSymbol:
+    private[tastyquery] def create(name: TermName, owner: Symbol): TermSymbol =
+      owner.addDeclIfDeclaringSym(TermSymbol(name, owner))
+  end TermSymbol
+
+  sealed abstract class TypeSymbol protected (val name: TypeName, override val owner: Symbol) extends Symbol(owner):
+    type ThisNameType = TypeName
+
+    // Overridden in TypeMemberSymbol
+    def isTypeAlias(using Context): Boolean = false
```

Comment:
I’d prefer a final method and pattern match

Reply:
Done.